### PR TITLE
Reset only when the operator value changed

### PR DESCRIPTION
**Pull request by devicedev**
https://etvas.atlassian.net/browse/ETV-10424

### CODE REVIEW

File: src/components/RuleBuilder/Rule.jsx [hunk 1]
```diff
@@ -44,29 +44,38 @@ export const Rule = ({
...
 
   const isSuffixType = useMemo(() => value.suffix, [value])
   const [, , { setValue: setRuleValue }] = useField(`${name}.${type}Value`)
 
   const onChangeRuleValue = useClearField([setRuleValue])
+  const [shouldReset, setShouldReset] = useState(false)
+
+  const handleOperatorChange = newValue => {
+    setOperatorValue(newValue)
+    setShouldReset(true)
```
Comment (ohnu) [inline src/components/RuleBuilder/Rule.jsx:70]: should the order be inversed ?
1. mark should reset true
2. change the value -> this triggers the effect

Comment (devicedev) [reply src/components/RuleBuilder/Rule.jsx:70]: I don't think so. The efect depends on the `shouldReset` state, not on the actual `operatorValue`. The setting of the `shouldReset` state is triggering the effect.

Comment (devicedev) [reply src/components/RuleBuilder/Rule.jsx:70]: There are two cases when the `operatorValue` can change.

1. The RuleBuilder is rendered initially, then the `operatorValue` receives a new value (from import)
2. The user selects a new `operatorValue` from the dropdown. ~ This is the case when we want to reset the state.

Comment (devicedev) [reply src/components/RuleBuilder/Rule.jsx]: f0f1c95d646aacf342452fcf346c45990638b44b

**Review by ohnu — APPROVED** [review]

**Pull request merged**